### PR TITLE
fix(Notification): refactor to not use caption as render conditional

### LINKED ISSUE #135: Refactor Notification to not rely on caption prop for rendering
Since `Notification` relies on the `caption` prop to render one of the notification variants: **Toast** or **Inline** notifications. 

But because of the work in #107, we need to provide a `defaultProp` for `caption` for `il8n`. 

This will result in `caption` always resolving as `true`... basically, the current Notifcation logic needs a refactor so that it's not relying on `caption` prop for rendering

### PR DESCRIPTION
**Pull request by hellobrian**
resolves #135

### CODE REVIEW

**Comment by marijohannessen** [comment]
👍 ✅

File: .storybook/components/NotificationStory.js [hunk 1]
```diff
@@ -10,45 +13,59 @@ const notificationProps = {
...
-    iconDescription: 'describes the close button',
-  },
+    iconDescription: 'describes the close button'
+  }
 };
 
 storiesOf('Notifications', module)
   .addWithInfo(
-    'toast',
+    'Deprecated: <Notfication />',
```
Comment (tw15egan) [inline .storybook/components/NotificationStory.js]: Notification*

**Comment by tw15egan** [comment]
👍 ✅

**Comment by chrisdhanaraj** [comment]
Bluemix [toolchain](https://console.bluemix.net/devops/toolchains/6b8b7163-1ec8-4616-82fa-aa467a17af03 "storybook-toolchain-v2"): [Delivery Pipeline](https://console.bluemix.net/devops/pipelines/b84aaac8-8484-4a1e-902a-ebb526eb7998 "storybook-toolchain-v2") deployed [carbon-components-react](https://github.com/carbon-design-system/carbon-components-react/releases/tag/deploy-ibm-yp-us-south-chrisdhanaraj_org-Bluemix-Design-System-20171011-005532 "deploy-ibm-yp-us-south-chrisdhanaraj_org-Bluemix-Design-System-20171011-005532") to [Bluemix Design System](https://console.bluemix.net/apps/bc75a101-6472-481c-b48d-3eed1e9703f3?env_id=ibm:yp:us-south "ibm:yp:us-south:chrisdhanaraj_org:Bluemix Design System"), including commit 9cb994c2710d8c63d2d97cd77c393781c869379b

**Pull request merged**